src/NetworkManager.h

```diff
@@ -0,0 +1,123 @@
+#ifndef NETWORK_MANAGER_H
+#define NETWORK_MANAGER_H
+
+#include "Preference.h"
+#include "StdString.h"
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include <ixwebsocket/IXHttp.h>
+#include <ixwebsocket/IXHttpClient.h>
+
+#include "EnumHelper.h"
+#include "LuaManager.h"
+
+struct lua_State;
+
+enum HttpErrorCode
+{
+	HttpErrorCode_Blocked,
+	HttpErrorCode_UnknownError,
+
+	// from IXWebSocket
+	HttpErrorCode_CannotConnect,
+	HttpErrorCode_Timeout,
+	HttpErrorCode_Gzip,
+	HttpErrorCode_UrlMalformed,
+	HttpErrorCode_CannotCreateSocket,
+	HttpErrorCode_SendError,
+	HttpErrorCode_ReadError,
+	HttpErrorCode_CannotReadStatusLine,
+	HttpErrorCode_MissingStatus,
+	HttpErrorCode_HeaderParsingError,
+	HttpErrorCode_MissingLocation,
+	HttpErrorCode_TooManyRedirects,
+	HttpErrorCode_ChunkReadError,
+	HttpErrorCode_CannotReadBody,
+	HttpErrorCode_Cancelled,
+
+	NUM_HttpErrorCode,
+	HttpErrorCode_Invalid,
+};
+const RString& HttpErrorCodeToString(HttpErrorCode dc);
+HttpErrorCode StringToHttpErrorCode(const RString& sDC);
+LuaDeclareType( HttpErrorCode );
+
+struct HttpRequestArgs
+{
+	std::string url;
+	std::string method = ix::HttpClient::kGet;
+	std::string body;
+	std::string multipartBoundary;
+	std::unordered_map<std::string, std::string> headers;
+	int connectTimeout = -1;
+	int transferTimeout = -1;
+	std::function<void(const ix::HttpResponsePtr& response)> onResponse = [](const ix::HttpResponsePtr& response) {};
+};
+
+class HttpRequestFuture
+{
+public:
+	HttpRequestFuture(ix::HttpRequestArgsPtr& args) : args(args) {};
+
+	static int gc(lua_State *L);
```

Comment:
What does gc stand for? GarbageCollect? Can we call it something more descriptive?

Reply:
Yes, that's what it stands for. I renamed it to `collect`.